src/Logger.js

```diff
@@ -192,12 +192,12 @@ function stringWidth(string) {
 // If we are in a worker, make a proxy class which will
 // send the logger calls to the main process via IPC.
 // These are handled in WorkerFarm and directed to handleMessage above.
-if (process.send && process.env.PARCEL_WORKER_TYPE === 'remote-worker') {
-  const worker = require('./worker');
+if (process.parcelWorker) {
```

Comment:
Maybe `WorkerFarm.isWorker` for this check? That could be set as a static property to equal `require.main === childModule` in the worker farm.